packages/sdk/src/agent.ts

```diff
@@ -0,0 +1,160 @@
+import {
+  ConnectionsModule,
+  CredentialsModule,
+  DidCommModule,
+  DiscoverFeaturesModule,
+  HttpOutboundTransport,
+  MediationRecipientModule,
+  MessagePickupModule,
+  OutOfBandModule,
+  ProofsModule,
+  WsOutboundTransport,
+} from '@credo-ts/didcomm'
+import { OpenId4VcHolderModule } from '@credo-ts/openid4vc'
+import { agentDependencies } from '@credo-ts/react-native'
+import { askar } from '@openwallet-foundation/askar-react-native'
+import { ParadymWalletMustBeDidCommAgentError, ParadymWalletMustBeOpenId4VcAgentError } from './error'
+import { type LogLevel, logger } from './logger'
+
+export type SetupAgentOptions = {
+  /**
+   *
+   * Unique identifier of your wallet storage
+   *
+   */
+  id: string
+
+  /**
+   *
+   * Key that will be used to securely open your wallet
+   *
+   * @security
+   * This property must dealt with in great care, such as receiving it from a biometrics check
+   *
+   */
+  key: string
+
+  /**
+   *
+   * The log level of the events that occur within the wallet
+   *
+   */
+  logLevel?: LogLevel
+
+  /**
+   *
+   * Configuration for when OpenId4Vc is used
+   *
+   * @note by default, openid4vc is configured on the agent
+   *
+   * @note to disable openid4vc, pass in `false`
+   *
+   */
+  openId4VcConfiguration?: ({} & X509ModuleConfigOptions) | false
+
+  /**
+   *
+   * Configuration for when DidComm is used
+   *
+   * @note by default, DIDComm is *not* configured on the agent
+   *
+   * @note to disable didcomm explicitly, pass in `false`
+   *
+   */
+  didcommConfiguration?:
+    | {
+        /**
+         *
+         * Label used for DIDComm connections to convey who you are to the other agent
+         *
+         */
+        label: string
+      }
+    | false
+}
+
+/**
+ *
+ * Function that sets up everything needed for the functionality required in a wallet.
+ *
+ */
+export const setupAgent = (options: SetupAgentOptions) => {
+  const openId4VcConfiguration = options.openId4VcConfiguration ?? {}
+
+  const didcommConfiguration = options.didcommConfiguration ?? false
+
+  const modules = {
+    askar: new AskarModule({
+      askar,
+    }),
+
+    ...(openId4VcConfiguration ? getOpenid4VcModules(openId4VcConfiguration) : {}),
+    ...(didcommConfiguration ? getDidCommModules(didcommConfiguration) : {}),
+  }
+
+  const agent = new Agent({
+    config: {
+      label: didcommConfiguration ? didcommConfiguration.label : '',
+      walletConfig: {
+        id: options.id,
+        key: options.key,
+        keyDerivationMethod: KeyDerivationMethod.Raw,
+      },
+      logger: options.logLevel ? logger(options.logLevel) : undefined,
+    },
+    dependencies: agentDependencies,
+    modules,
+  })
+
+  if (didcommConfiguration) {
+    ;(agent as DidCommAgent).modules.didcomm.registerOutboundTransport(new HttpOutboundTransport())
+    ;(agent as DidCommAgent).modules.didcomm.registerOutboundTransport(new WsOutboundTransport())
+  }
```

Comment:
I can't handle this syntax

```suggestion
  if (didcommConfiguration) {
    const didcommAgent  = agent as DidCommAgent
    didcommAgent.modules.didcomm.registerOutboundTransport(new HttpOutboundTransport())
    didcommAgent.modules.didcomm.registerOutboundTransport(new WsOutboundTransport())
  }
```

Reply:
This is the exact code I had before, but it does require an allocation which I would like to avoid as it is just about the typing. I agree `;()` is quite ugly.

Reply:
Hmm.  I'm always of the opinion "clean code > optimized code" until performance becomes an issue, but fine by me